src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractTransactionService.cs

```diff
@@ -415,18 +415,21 @@ public List<ReceiptResponse> ReceiptSearch(List<string> contractAddresses, strin
         {
             var filteredContractAddresses = new HashSet<string>();
 
-            foreach (string contractAddress in contractAddresses)
+            if (contractAddresses != null)
```

Comment:
Can this list be null?

Reply:
If it is passed as null or empty, then the assumption is that all contracts should be searched for matching logs